cmd/oras/internal/errors/errors.go

```diff
@@ -60,6 +65,45 @@ func CheckArgs(checker func(args []string) (bool, string), Usage string) cobra.P
 	}
 }
 
+// Handler handles error during cmd execution.
+type Handler interface {
+	// Handle handles error during cmd execution.
+	// if nil is returned, then err cannot be handled.
+	Handle(err error, cmd *cobra.Command) error
+}
+
+// Command returns an error-handled cobra command.
+func Command(cmd *cobra.Command, handler Handler) *cobra.Command {
+	runE := cmd.RunE
+	cmd.RunE = func(cmd *cobra.Command, args []string) error {
+		err := runE(cmd, args)
+		if err == nil {
+			return nil
+		}
+		if handled := handler.Handle(err, cmd); handled != nil {
+			return handled
+		}
+		return err
```

Comment:
```suggestion
		err := runE(cmd, args)
		if err != nil {
			return handler.Handle(err, cmd)
		}
		return nil
```

Reply:
In fact, this handler is not a handler but a modifier.

Therefore, consider
```go
type Modifier interface {
    Modify(cmd *cobra.Command, err error) error
}

Reply:
> return handler.Handle(err, cmd)

This is wrong, if the function returns nil, then the original `err` must be returned.